### DISCUSSION TITLE
How to read map files? Defining an array bigger than 108 breaks my game.

### DISCUSSION THREAD

**Discussion by Fabrizio-Caruso** [cc65/cc65#1520] [Q&A]
In the last days I am trying to understand how to use the Apple2 hgr cfg. I have tried to follow some instructions and the doc. I have tried to read the map file to see what is wrong.

What I see is that if I define an array variable 
`uint8_t hgr_enc[] = { ... } `
my game will just crash at start-up unless the array size is small enough. No matter what I do with the array (I do nothing with it).
Size 108 is ok, while any size bigger than that, breaks the game. How can I debug this? I do not see any information about hgr_enc in the map file.

Is there a limit to the size of the initialization of an array?

The map file does not show any evident error to me but I am not sure I know how to interpret it:
```
Name                   Start     End    Size  Align
----------------------------------------------------
ZEROPAGE              000080  000099  00001A  00001
EXEHDR                0007C9  000802  00003A  00001
STARTUP               000803  000836  000034  00001
LOWCODE               000837  000B7A  000344  00001
HGR                   002000  003FFF  002000  00001
CODE                  004000  007D54  003D55  00001
RODATA                007D55  007EFA  0001A6  00001
DATA                  007EFB  00816D  000273  00001
INIT                  00816E  00818A  00001D  00001
BSS                   00818B  008303  000179  00001
ONCE                  00818B  008200  000076  00001
```

What else can I do to debug this?

**Comment by groessler** [comment]
I think LOWCODE must be filled to 0x2000. Simplest is to use "fill" option in .cfg file.

**Reply by Fabrizio-Caruso** [reply]
How can I do this on a segment? I think I have done it on memory areas but never on segments. 
The map file seems to say CODE is starting at the right address.
@greg-king5 wrote the linker pads LOWCODE automatically.

**Comment by Fabrizio-Caruso** [comment]
I am not sure but MAIN may be taking up the whole available memory.
If I compile with -O, I get a working game with no sound because -O destroys any code (C or Assembly) that is just peeking at one location and doing nothing else with it (I need it to produce sounds through bit-banging).

**Reply by groessler** [reply]
Ignore my answer. It gets auto-filled.

**Reply by greg-king5** [reply]
> I am not sure, but MAIN may be taking up the whole available memory.

It looks small enough, to me.
$8303 is the highest end address in that segment list.  Adding the C stack's size gives $8B03.
You seem to have plenty of room.

**Comment by greg-king5** [comment]
You should be able to initiate both global and static arrays that are as big as 64 Kibibytes.  Even an "automatic" array (on the C stack) should be able to have a size that's bigger than 108!

A standard map file doesn't mention names that aren't imported.  (You didn't _use_ that array anywhere.)  You must add the option `-vm` to your command line when you want to see everything that is exported.
